src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -18,7 +19,23 @@ using namespace Microsoft::Console::Render;
 using namespace Microsoft::Console::Types;
 using namespace Microsoft::Console::VirtualTerminal;
 
-std::wstring _KeyEventsToText(std::deque<std::unique_ptr<IInputEvent>>& inEventsToWrite)
+static short _ClampToShortMax(int value, short min)

```

Comment:
I'm not sure that you need to introduce this function. There might be an easier way to do this with `gsl::narrow<short>()`, but it might need to be something like
```c++
short result = 1;
try
{
    result = gsl::narrow<short>(value);
} CATCH_LOG();

@miniksa might know a simpler way.
```

Reply:
Why not just std::clamp(value, 1, SHRT_MAX)?

Reply:
I was actually hoping to use std::clamp(), but I didn't see clamp() in [the MSDN doc page for &lt;algorithm&gt;](https://docs.microsoft.com/cpp/standard-library/algorithm-functions?view=vs-2019), so I didn't realize [Microsoft's STL supports clamp()](https://docs.microsoft.com/cpp/overview/visual-cpp-language-conformance?view=vs-2019)! I opened documentation issue https://github.com/MicrosoftDocs/cpp-docs/issues/1010 for this.

Unfortunately clamp() introduces other problems. We need to be careful to clamp the 32-bit integers first into a range that fits in 16 bits and _then_ stuff the result into 16 bits, otherwise we reintroduce the overflow bug. Since all the values we want to clamp are 32-bit `int` values, we end up writing things like:

```c++
Create(viewportSize, static_cast<short>(std::clamp(settings.HistorySize(), 0, SHRT_MAX)), renderTarget);
```

which IMO is a little less readable than what I wrote:

```c++
Create(viewportSize, _ClampToShortMax(settings.HistorySize(), 0), renderTarget);
```

I don't feel strongly enough to stand firm on this, but for now I haven't removed the helper function but instead turned it into a wrapper for clamp():

```c++
short _ClampToShortMax(int value, short min) {
  return static_cast<short>(std::clamp(value, static_cast<int>(min), SHRT_MAX));
}
```

Let me know if you want me to call clamp() directly and I'll make that change.

Reply:
You make a good point.

I'm almost thinking you should throw this into a utility function inside the `types` library so it's available everywhere and turn it into a template of the type of

```
T _ClampTo<T>(U value, T min)
```

Maybe that's overkill? I just see it being useful.

Reply:
Sure, I'll do something like what you suggested.

Reply:
While I tried, it seems writing C++ templates is just too hard for me to do correctly and readably. So is it ok to keep this as it currently is? We can always extract the function out into a common header, with or without templatization, when it becomes necessary do so.

Reply:
Yeah it's fine. Just an idea.